pkg/vsphere/disk/lazy_detach_test.go

```diff
@@ -0,0 +1,185 @@
+		if err != nil && err.Error() == "can't find the hosting vm" {
+			t.Skip("Skipping: test must be run in a VM")
+		}
+		if !assert.NoError(t, err) {
+			return
+		}
+		_, err = task.WaitForResult(context.TODO(), nil)
+		if !assert.NoError(t, err) {
+			return
+		}
+	}()
+
+	op := trace.NewOperation(context.TODO(), "lazyReconfigure")
+
+	// create a diskmanager
+	vdm, err := NewDiskManager(op, client)
+	if err != nil && err.Error() == "can't find the hosting vm" {
+		t.Skip("Skipping: test must be run in a VM")
+	}
+	if !assert.NoError(t, err) || !assert.NotNil(t, vdm) {
+		return
+	}
+
+	// helper fn
+	reconfigure := func(changes []types.BaseVirtualDeviceConfigSpec) error {
+		t.Logf("Calling reconfigure")
+
+		machineSpec := types.VirtualMachineConfigSpec{}
+		machineSpec.DeviceChange = changes
+
+		_, err := vdm.vm.WaitForResult(op, func(ctx context.Context) (tasks.Task, error) {
+			t, er := vdm.vm.Reconfigure(ctx, machineSpec)
+
+			if t != nil {
+				op.Debugf("reconfigure task=%s", t.Reference())
+			}
+
+			return t, er
+		})
+		return err
+	}
+
+	// 1MB
+	diskSize := int64(1 << 10)
+	scratch := &object.DatastorePath{
+		Datastore: client.Datastore.Name(),
+		Path:      path.Join(imagestore.Path, "scratch.vmdk"),
+	}
+	// config
+	config := NewPersistentDisk(scratch).WithCapacity(diskSize)
+
+	// attach + create spec (scratch)
+	spec := []types.BaseVirtualDeviceConfigSpec{
+		&types.VirtualDeviceConfigSpec{
+			Device:        vdm.toSpec(config),
+			Operation:     types.VirtualDeviceConfigSpecOperationAdd,
+			FileOperation: types.VirtualDeviceConfigSpecFileOperationCreate,
+		},
+	}
+	// reconfigure
+	err = reconfigure(spec)
+	if !assert.NoError(t, err) {
+		return
+	}
+	t.Logf("scratch created and attached")
+
+	// ref to scratch (needed for detach as initial spec's Key and UnitNumber was unset)
+	disk, err := findDiskByFilename(op, vdm.vm, scratch.Path)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// DO NOT DETACH AND START WORKING ON THE CHILD
+
+	// child
+	child := &object.DatastorePath{
+		Datastore: client.Datastore.Name(),
+		Path:      path.Join(imagestore.Path, "child.vmdk"),
+	}
+	// config
+	config = NewPersistentDisk(child).WithParent(scratch)
+
+	// detach (scratch) AND attach + create (child) spec
+	spec = []types.BaseVirtualDeviceConfigSpec{
+		&types.VirtualDeviceConfigSpec{
+			Device:    disk,
+			Operation: types.VirtualDeviceConfigSpecOperationRemove,
+		},
+		&types.VirtualDeviceConfigSpec{
+			Device:        vdm.toSpec(config),
+			Operation:     types.VirtualDeviceConfigSpecOperationAdd,
+			FileOperation: types.VirtualDeviceConfigSpecFileOperationCreate,
+		},
+	}
+	// reconfigure
+	err = reconfigure(spec)
+	if !assert.NoError(t, err) {
+		return
+	}
+	t.Logf("scratch detached, child created and attached")
+
+	// ref to child (needed for detach as initial spec's Key and UnitNumber was unset)
+	disk, err = findDiskByFilename(op, vdm.vm, child.Path)
+	if !assert.NoError(t, err) {
+		return
+	}
+
```

Comment:
check for the file created on scratch here - this already validates that the combined remove/add operation is permitted, but doesn't confirm actual inheritance.